### PR TITLE
Tracking the Pay button in the checkout flow. 

### PR DESCRIPTION
**Pull request by svillafe**
## Why are we doing this? 

 This work is part of a three cards task. We first want to track the **Pay** button so in the near future we are able to implement an A/B Test to test the different button's text.

## Which button are we tracking?
We are now tracking the **Pay** button on the checkout page. 

track-clicks-pay-button

## What are we sending to GA?

The following data that is being sent to GA in the events:

### Flow A

**data-metric-trigger**="click"
**data-metric-category**="checkout page"
**data-metric-action**="cta click"
**data-metric-label**="payment a"

### Flow B

**data-metric-trigger**="click"
**data-metric-category**="checkout page"
**data-metric-action**="cta click"
**data-metric-label**="payment b"

## Related Trello cards:
**"Capture the price"**: [card: 172](https://trello.com/c/cyKgv9eD/172-capture-the-price-that-someone-is-paying)
**"A/B Test"**: [card: 167](https://trello.com/c/cQn4Fixp/167-a-b-test-different-prices-on-the-cta)
**"Consistent journey"**: [card: 189](https://trello.com/c/9mTB8UAR/189-keep-pricing-throughout-the-journey-consistent)

### CODE REVIEW

File: frontend/app/views/fragments/form/submit.scala.html [hunk 1]
```diff
@@ -37,7 +39,14 @@
             <noscript>
                 <a href="http://www.enable-javascript.com/" target="_blank" class="action is-disabled">Enable javascript to proceed</a>.
             </noscript>
-            <button type="button" class="action js-submit-input u-hidden-non-js">
+            <button type="button" class="action js-submit-input u-hidden-non-js"
+                    @if(dataCategory != None && dataLabel != None) {
```
Comment (rtyley) [inline frontend/app/views/fragments/form/submit.scala.html]: If you see `if` statements and `Option`s together, you can usually guess there is a more scala-ish way of doing things! eg, use a for-comprension:

```
@for(
  label <- dataLabel
  category <- dataCategory
) {
  data-metric-category=@category
  data-metric-label=@dataLabel
}
```

**Comment by rtyley** [comment]
Hi @svillafe - could you link this to a trello story? Can you add a bit of the 'why' - ie why are we adding this?

File: frontend/app/views/fragments/form/submit.scala.html (hunk 1 above)
Comment (svillafe) [inline frontend/app/views/fragments/form/submit.scala.html]: Originally I tried to implemented with a for expr (for-yield) but that kind of expression is not supported by the scala template engine. The way that you suggested above is not supported either. 

The only way to do this using a for expr is: 

```
@for(
  label <- dataLabel
) {
  data-metric-label=label
}
@for(
  category <- dataCategory
) {
  data-metric-category=category
}
```

**Comment by svillafe** [comment]
@rtyley Updated :)

File: frontend/app/views/fragments/form/submit.scala.html (hunk 1 above)
Comment (rtyley) [inline frontend/app/views/fragments/form/submit.scala.html]: Apologies, for Twirl templates the syntax looks more like this:

```
@for(label <- dataLabel; category <- dataCategory) {
  data-metric-category=@category
  data-metric-label=@dataLabel
}
```

...there's an existing example here: https://github.com/guardian/membership-frontend/blob/cb9cf5bfce/frontend/app/views/fragments/content/articleSnapshot.scala.html#L11

**Comment by rtyley** [comment]
:+1:

**Comment by svillafe** [comment]
It is working on PROD

**Comment by johnduffell** [comment]
this is weird, prout put your post deployment test retry (that failed) on my PR that should have passed, maybe because your retry happened after my changes were merged @rtyley https://github.com/guardian/membership-frontend/pull/1397#issuecomment-264430312

**Pull request merged**